actionview/lib/action_view/record_identifier.rb

```diff
@@ -103,8 +103,9 @@ def dom_id(record, prefix = nil)
     # overwritten version of the method. By default, this implementation passes the key string through a
     # method that replaces all characters that are invalid inside DOM ids, with valid ones. You need to
     # make sure yourself that your dom ids are valid, in case you override this method.
-    def record_key_for_dom_id(record) # :doc:
-      key = convert_to_model(record).to_key
+    def record_key_for_dom_id(record_or_class) # :doc:
+      model = convert_to_model(record_or_class)
+      key = model.respond_to?(:to_key) ? model.to_key : nil
```

Comment:
I'm on the fence about this method implementation since it can only work for a model and never for a class. I feel we shouldn't even call this method if we have a class rather than having to twice check our value and return `nil` if it's a class.

Reply:
I've updated the logic in `dom_id` following your feedback, is it better now?

```ruby
def dom_id(record_or_class, prefix = nil)
  record_id = record_key_for_dom_id(record_or_class) unless record_or_class.is_a?(Class)
  if record_id
    "#{dom_class(record_or_class, prefix)}#{JOIN}#{record_id}"
  else
    dom_class(record_or_class, prefix || NEW)
  end
end
```